test/aerospike_clj/client_test.clj

```diff
@@ -69,14 +69,74 @@
     (testing "clojure maps can be serialized as-is")
     (let [v @(client/get-single-no-meta *c* K _set)]
       (is (= data v)) ;; per value it is identical
-      (is (= java.util.HashMap (type v))) ;; but we get back a HashMap
-      (is (false? (map? v)))
+      (is (= clojure.lang.PersistentArrayMap (type v)))
       (testing "using Jackson to recursively create a PersistantHashMap"
         (let [json (.writeValueAsString (ObjectMapper.) v)
               parsed (json/parse-string json #(keyword (subs % 1)))]
           (is (= data parsed))
           (is (true? (map? parsed))))))))
 
+(deftest put-multiple-bins-get-clj-map
+  (let [data {:foo {:bar [(rand-int 1000)]}
+              :baz true
+              :qux false
+              :quux nil}]
+    (is (true? @(client/create *c* K _set data 100)))
+    (testing "clojure maps can be serialized from bins")
+    (let [v @(client/get-single-no-meta *c* K _set)]
+      (is (= (:foo data) (:foo v))) ;; per value it is identical
+      (is (= (:baz data) (:baz v))) ;; true value returns the same after being sanitized/desanitized
+      (is (= (:qux data) (:qux v))) ;; false value returns the same after being sanitized/desanitized
+      (is (= (:quux data) (:quux v))) ;; nil value retuns the same after being sanitized/desanitized
+      (is (= clojure.lang.PersistentArrayMap (type v))) ;; converted back to a Clojure map instead of HashMap
+      (is (true? (map? v)))
+      (testing "using Jackson to recursively create a PersistantHashMap"
+        (let [json (.writeValueAsString (ObjectMapper.) v)
+              parsed (json/parse-string json #(keyword (subs % 1)))]
+          (is (= data parsed))
+          (is (true? (map? parsed))))))))
+
+(deftest get-single-multiple-bins
+  (let [data {:foo [(rand-int 1000)]
+              :bar [(rand-int 1000)]
+              :baz [(rand-int 1000)]}]
+    (is (true? @(client/create *c* K _set data 100)))
+    (testing "bin values can be retrieved individually and all together")
+    (let [v1 @(client/get-single *c* K _set [:foo])
+          v2 @(client/get-single *c* K _set [:bar])
+          v3 @(client/get-single *c* K _set [:baz])
+          v4 @(client/get-single *c* K _set [:all])]
+      (is (= (:foo data) (:foo (:payload v1))))
+      (is (= (:bar data) (:bar (:payload v2))))
+      (is (= (:baz data) (:baz (:payload v3))))
+      (is (= data (:payload v4)))
+      (is (true? (map? (:payload v1)))))))
+
+(deftest adding-bins-to-record
+  (let [data {:foo [(rand-int 1000)]
+              :bar [(rand-int 1000)]
+              :baz [(rand-int 1000)]}
+        new-data {:qux [(rand-int 1000)]}]
+    (is (true? @(client/create *c* K _set data 100)))
+    (is (true? @(client/add-bins *c* K _set new-data 100))) ;; adding value to bin
+    (testing "bin values can be added to existing records")
+    (let [v @(client/get-single-no-meta *c* K _set)]
+      (is (= v (merge data new-data)))
```

Comment:
maybe also test what happens if you call `add-bins` with an existing bin?

Reply:
:+1: